pkg/reconciliation/construct_service.go

```diff
@@ -78,7 +79,13 @@ func addAdditionalOptions(service *corev1.Service, serviceConfig *api.ServiceCon
 			service.Labels = make(map[string]string, len(serviceConfig.Labels))
 		}
 		for k, v := range serviceConfig.Labels {
-			service.Labels[k] = v
+			// make sure user specified labels don't override reserved labels
+			if oplabels.IsReservedLabel(k) || api.IsReservedLabel(k) {
+				// don't allow reserved labels as AdditionalLabels
+				log.Log.Info("Skipping reserved label from AdditionalLabels", "reserved label", k)
```

Comment:
@jsanda Not sure if this is the desired behavior, but I implemented a log and just ignoring any user-supplied labels/annotations that conflict with reserved ones

Reply:
Just don't let user to include anything with `cassandra.datastax.com` prefix should be enough. Or even `k8ssandra.io` for that matter.

Reply:
For both labels and annotations I presume?

Reply:
Pardon the intrusion... Should we raise a ticket to implement logic in a webhook to reject resources which use our labels? 

Labels can be critical to integrations in k8s, so if a label gets ignored without the user realising the blast radius for unexpected behaviour can be wide.

Reply:
> For both labels and annotations I presume?

Yes. I think we mentioned somewhere that those are reserved for internal use.

Reply:
> Pardon the intrusion... Should we raise a ticket to implement logic in a webhook to reject resources which use our labels? 

I don't think it needs another ticket necessarily if Erik has time to implement it here. Just add it to the bottom of the current webhook (put the validation logic in the api package).